app/models/import_jobs.py

```diff
@@ -53,7 +53,12 @@ def import_jobs(location):
                 if os.path.isdir(potential_job):
                     job_name = os.path.basename(potential_job)
                     logger.debug('       job {}'.format(job_name))
-                    job, added = add_job(potential_job, job_name, project)
+                    try:
+                        job, added = add_job(potential_job, job_name, project)
+                    except sqlite3.IntegrityError as e:
```

Comment:
I think we do not want to introduce `sqlite3` here. The dashboard is written to work with `SQLAlchemy`. We think in most cases, people will want to use `sqlite`, but with `sqlalchemy` we have the flexibility for that to not be the case. Introducing the use of `sqlite3` will restrict the use of this script to sqlite databases.

Reply:
Three points: first is that I am trying to figure out how to integrate with the Dashboard, in which case it would use the same db instance, etc. Second, the database access and SQL it uses is extremely simple, so there is no issue using and maintaining a separate approach, and integrating SQLAlchemy was actually not so easy (at least for me) And third, maybe the best approach is to not touch the database at all in the JobServer, but use the Daskboard's API and make http calls.

At this point it works sufficiently, so I think we use as is and concentrate on being able to run jobs elsewhere, catching the various errors that will happen, etc. and let this "mature". In a month or so we will have a much better understanding of how to tie this to the Dashboard, what the requirements are and the issues. We may also find it depends on bit on whether we use Flask or a production server, etc.